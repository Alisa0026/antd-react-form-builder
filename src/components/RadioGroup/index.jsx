import React from 'react'
import _ from 'lodash'
import moment from 'moment'
import BasicItem from '../BasicItem'
import {
  Radio 
} from "antd"
import FormItemComponentDecorator from '../../decorator/FormItemComponent'

const RadioGroup = Radio.Group;

function component(BasicItemComponent){
  @FormItemComponentDecorator
  class FRadioGroup extends React.Component {

    render(){
      let { options,...other } = this.props;
      other.targetComponent = RadioGroup;
      other.type = "radiogroup";
      this.propsAdapter(other);
      return (
        <BasicItemComponent { ...other }>
          {
            options && options[0] && options.map((v,k)=>{
              return (
                <Radio value={ v.value } key={ k }>
                  { v.label }
                </Radio>
              )
            })
          }
        </BasicItemComponent>
      )
    }
    
  }
  return FRadioGroup;
}

export default component(BasicItem) 




