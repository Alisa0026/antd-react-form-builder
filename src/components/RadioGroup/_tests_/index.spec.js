import React from 'react'
import { shallow,mount,render } from 'enzyme'
import sinon from 'sinon'
import LocaleProvider from "antd/lib/locale-provider"
import { 
  testCommonProps, 
  getAntdFormComponentMount,
  getAntdFormItemMount,
} from '../../../../tests/common'
import RadioGroup from '../index'

var options = [
  {
    value: "shenzhen",
    label: "深圳"
  },
  {
    value: "beijing",
    label: "北京"
  },
  {
    value: "hangzhou",
    label: "hangzhou"
  },
];

describe('RadioGroup',function(){
  sinon.stub(console,"warn");
  it('renders correctly', () => {
    const wrapper = render(
      <RadioGroup 
        required
        className="test"
        label="name"
        name="test"
        options={ options }
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  testCommonProps(RadioGroup); 
  it("should have three <Radio />",function(){
    const wrapper = mount(
      <RadioGroup 
        required
        name="test"
        label="label"
        options={ options }
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    expect(antdInputMount.props().children.length).toEqual(3);
  })
})

