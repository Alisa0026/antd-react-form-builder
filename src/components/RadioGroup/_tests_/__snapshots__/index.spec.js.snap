// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup renders correctly 1`] = `
<span>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-form-item-label"
    >
      <label
        class="ant-form-item-required"
        title="name"
      >
        name
      </label>
    </div>
    <div
      class="ant-form-item-control-wrapper"
    >
      <div
        class="ant-form-item-control "
      >
        <div
          class="ant-radio-group ant-radio-group-default test"
        >
          <label
            class="ant-radio-wrapper"
          >
            <span
              class="ant-radio"
            >
              <input
                class="ant-radio-input"
                type="radio"
              />
              <span
                class="ant-radio-inner"
              />
            </span>
            <span>
              深圳
            </span>
          </label>
          <label
            class="ant-radio-wrapper"
          >
            <span
              class="ant-radio"
            >
              <input
                class="ant-radio-input"
                type="radio"
              />
              <span
                class="ant-radio-inner"
              />
            </span>
            <span>
              北京
            </span>
          </label>
          <label
            class="ant-radio-wrapper"
          >
            <span
              class="ant-radio"
            >
              <input
                class="ant-radio-input"
                type="radio"
              />
              <span
                class="ant-radio-inner"
              />
            </span>
            <span>
              hangzhou
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
  <input
    class="ant-input"
    name="test"
    type="hidden"
    value=""
  />
</span>
`;
